src/index.ts

```diff
@@ -401,6 +410,20 @@ const ceil = (x1: string, precision: number): string =>
 const round = (x1: string, precision: number): string =>
   precisionAdjust('round', x1, precision)
 
+function toBns(n: number | string): string {
+  let out = typeof n === 'number' ? n.toString() : n
+  // Handle scientific notation
+  const match = out.match(SCI_NOTATION_REGEX)
+
+  if (match != null) {
+    const exponent = parseInt(match[1])
+    const base = out.replace(SCI_NOTATION_REGEX, '')
```

Comment:
Replace with regex with `SCI_NOTATION_REGEX = /(.[0-9]*)e((?:\+|-)\d+)$/` Then you only have to apply it once to get both the exponent and base.